arbor/communication/communicator.cpp

```diff
@@ -33,6 +34,12 @@ communicator::communicator(const recipe& rec,
     num_local_groups_ = dom_dec.groups.size();
     num_local_cells_ = dom_dec.num_local_cells;
 
+    std::vector<cell_size_type> cell_num_sources;
```

Comment:
Storing these for all cells in the model will use a lot of memory for large models. Can we just look these up at runtime, and optimize it later if need be?

Reply:
I thought it might make more sense to do it this way cause otherwise we're calling `num_sources(gid)` once for every connection, and there's typically many more connections  than cells. But I can undo it.

Reply:
It's a trade off: and which approach is best will depend on the model.
I assumed that the vast majority of models will automatically return 1 for `num_sources`, calculating the value on the fly would be cheap... but worst case scenario they generate the cell every time they ask the question.

Reply:
I agree. It makes more sense to save the memory.